stl/inc/numeric

```diff
@@ -461,6 +461,20 @@ _FwdIt2 transform_inclusive_scan(_ExPo&& _Exec, _FwdIt1 _First, _FwdIt1 _Last, _
     _UnaryOp _Transform_op) noexcept; // terminates
 #endif // _HAS_CXX17
 
+template <class _TyDest, class _TySrc, class _BinOp>
+void _Adjacent_difference_no_overlap(
+    _TyDest* const __restrict _Dest, _TySrc* const __restrict _Src, const size_t _Count, _BinOp _Func) {
+    _Dest[0] = _Src[0];
+    for (size_t _Ix = 1; _Ix != _Count; ++_Ix) {
+#if _HAS_CXX20
+        _TySrc _Tmp = _Src[_Ix - 1];
+        _Dest[_Ix]  = _Func(_Src[_Ix], _STD move(_Tmp));

```

Comment:
You've checked `is_trivially_copyable_v<_Iter_value_t<_InIt>>` , so I don't see how this `move` could be observable.

Reply:
Unlike other vectorized algorithms, this one still calls potentially user defined predicate. it might have the lvalue overload available, but not accessible.

I would not have imagined that, but llvm tests have dedicated coverage.

Reply:
Oh, I see, `_Func` could care about the value category, got it.